packages/components/src/components/text-field/text-field.tsx

```diff
@@ -156,6 +168,11 @@ export class TextField {
 
   handleInput = (event: Event) => {
     const target = event.target as HTMLInputElement | null;
+    if (this.controlled) {
+      emitEvent(this, 'scaleChange', { value: target.value });
```

Comment:
Is firing `scaleChange` here really needed? it's being fired again on line 178. Also, doesn't the re-render triggered by `forceUpdate` fire another change event?

Maybe it'd make sense to add a test to check whether the change event is not being fired too much?

Reply:
yup that was a mistake, removed and added a test

Reply:
amazing, thank you!